webgoat-lessons/path-traversal/src/main/java/org/owasp/webgoat/path_traversal/ProfileUploadBase.java

```diff
@@ -0,0 +1,93 @@
+package org.owasp.webgoat.path_traversal;
+
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import org.owasp.webgoat.assignments.AssignmentEndpoint;
+import org.owasp.webgoat.assignments.AttackResult;
+import org.owasp.webgoat.session.WebSession;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.util.FileSystemUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Base64;
+
+@AllArgsConstructor
+public class ProfileUploadBase extends AssignmentEndpoint {
+
+    private String webGoatHomeDirectory;
+    private WebSession webSession;
+
+    protected AttackResult execute(MultipartFile file, String fullName) {
+        if (file.isEmpty()) {
```

Comment:
both are successful, the code does try to create a new file first which fails if the file already exists this will safeguard when one tries to overwrite an existing file

Reply:
fixed